File: be/src/users/dto/update-user.dto.ts
import { Type } from 'class-transformer';
import { IsDate, IsEnum, IsOptional, IsString } from 'class-validator';

export class UpdateUserDto {
  @IsOptional()
  @IsString()
  name?: string;

  @IsOptional()
  @Type(() => Date)
  @IsDate()
  dateOfBirth?: Date;

  @IsOptional()
  @IsEnum(['male', 'female', 'other'])
  gender?: 'male' | 'female' | 'other'; 

  @IsOptional()
  @IsString()
  studentCode?: string;

  @IsOptional()
  @IsString()
  department?: string;
}

File: backend-java/src/main/java/com/qlda/backend/users/dto/UpdateUserDto.java
package com.qlda.backend.users.dto;

import jakarta.validation.constraints.Pattern;
import lombok.Data;

import java.time.LocalDate;

@Data
public class UpdateUserDto {
    private String name;
    private LocalDate dateOfBirth;
    
    @Pattern(regexp = "male|female|other", message = "Gender phải là male, female hoặc other")
    private String gender;
    private String studentCode;
    private String department;
}

